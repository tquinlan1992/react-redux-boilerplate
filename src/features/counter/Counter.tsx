import React, { useState } from "react";

import { useAppSelector, useAppDispatch } from "../../app/hooks";
import {
  decrement,
  increment,
  incrementByAmount,
  incrementAsync,
  incrementIfOdd,
  selectCount,
} from "./counterSlice";
import styles from "./Counter.module.css";

export function Counter() {
  const count = useAppSelector(selectCount);
  const dispatch = useAppDispatch();
  const [incrementAmount, setIncrementAmount] = useState("2");

  const incrementValue = Number(incrementAmount) || 0;

  return (
    <CounterPresentational
      count={count}
      onDecrementClick={() => dispatch(decrement())}
      onIncrementClick={() => dispatch(increment())}
      incrementAmount={incrementAmount}
      incrementValue={incrementValue}
      setIncrementAmount={setIncrementAmount}
      incrementByAmount={() => dispatch(incrementByAmount(incrementValue))}
      incrementAsync={() => dispatch(incrementAsync(incrementValue))}
      incrementIfOdd={() => dispatch(incrementIfOdd(incrementValue))}
    />
  );
}

export const CounterPresentational: React.VFC<{
  count: number;
  onDecrementClick: () => void;
  onIncrementClick: () => void;
  incrementAmount: string;
  incrementValue: number;
  setIncrementAmount: (value: string) => void;
  incrementByAmount: (value: number) => void;
  incrementAsync: (value: number) => void;
  incrementIfOdd: (value: number) => void;
}> = ({
  onDecrementClick,
  onIncrementClick,
  count,
  incrementAmount,
  setIncrementAmount,
  incrementByAmount,
  incrementValue,
  incrementAsync,
  incrementIfOdd,
}) => (
  <div>
    <div className={styles.row}>
      <button
        className={styles.button}
        aria-label="Decrement value"
        onClick={onDecrementClick}
      >
        -
      </button>
      <span className={styles.value}>{count}</span>
      <button
        className={styles.button}
        aria-label="Increment value"
        onClick={onIncrementClick}
      >
        +
      </button>
    </div>
    <div className={styles.row}>
      <input
        className={styles.textbox}
        aria-label="Set increment amount"
        value={incrementAmount}
        onChange={(e) => setIncrementAmount(e.target.value)}
      />
      <button
        className={styles.button}
        onClick={() => incrementByAmount(incrementValue)}
      >
        Add Amount
      </button>
      <button
        className={styles.asyncButton}
        onClick={() => incrementAsync(incrementValue)}
      >
        Add Async
      </button>
      <button
        className={styles.button}
        onClick={() => incrementIfOdd(incrementValue)}
      >
        Add If Odd
      </button>
    </div>
  </div>
);
