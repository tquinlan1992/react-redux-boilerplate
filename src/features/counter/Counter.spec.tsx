import React from "react";
import renderer from "react-test-renderer";
import { CounterPresentational, CounterPresentationalProps } from "./Counter";

it("renders correctly", () => {
  const props: CounterPresentationalProps = {
    count: 2,
    incrementAmount: "1",
    incrementValue: 0,
    onDecrementClick: jest.fn(),
    onIncrementClick: jest.fn(),
    setIncrementAmount: jest.fn(),
    incrementByAmount: jest.fn(),
    incrementAsync: jest.fn(),
    incrementIfOdd: jest.fn(),
  };
  const tree = renderer.create(<CounterPresentational {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});
