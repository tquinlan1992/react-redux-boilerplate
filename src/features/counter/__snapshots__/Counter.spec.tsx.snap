// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    className="row"
  >
    <button
      aria-label="Decrement value"
      className="button"
      onClick={[MockFunction]}
    >
      -
    </button>
    <span
      className="value"
    >
      2
    </span>
    <button
      aria-label="Increment value"
      className="button"
      onClick={[MockFunction]}
    >
      +
    </button>
  </div>
  <div
    className="row"
  >
    <input
      aria-label="Set increment amount"
      className="textbox"
      onChange={[Function]}
      value="1"
    />
    <button
      className="button"
      onClick={[Function]}
    >
      Add Amount
    </button>
    <button
      className="asyncButton"
      onClick={[Function]}
    >
      Add Async
    </button>
    <button
      className="button"
      onClick={[Function]}
    >
      Add If Odd
    </button>
  </div>
</div>
`;
